server/pulp/server/webservices/views/util.py

```diff
@@ -61,7 +82,8 @@ def wrapper(*args, **kwargs):
             return func(*args, **kwargs)
 
         try:
-            request.body_as_json = json.loads(request.body)
+            request_json = json.loads(request.body)
+            request.body_as_json = _ensure_input_encoding(request_json)
```

Comment:
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA1

Even worse is catching an Exception, and then transforming it into a
PulpSomethingSomethingException and raising that, but trying to keep
the same traceback and other frame information. This happened to me
while trying to hunt down a very mysterious test failure last night,
and it was the worst. The Exeption that actually made it to the top
had no information to help me debug, because it was all hidden by that
bit that transformed it. Unfortunately, this pattern happens all over
our code from the old days ☹
-----BEGIN PGP SIGNATURE-----
Version: GnuPG v1

iQIcBAEBAgAGBQJU5NpfAAoJEIyFaKUJtmpi8t4P/0ZKfemOuIkxSmqMz2hnvJrs
km0LOzMKassFu6NIosc161/DbaIU2tuRYEhuzR5wJh7H0N0/Qijnc80TjvLFpMZQ
pzg4+eneUsL3E5ahTPOP/SdlIOPxCU76IKtXLRPWynQ4oG3MRufuPBkvFYHUHoVY
CHQuTN07bwSWj/SBlaxhFbrhZOe8QKDf8XtuPVMP3JnHedC5d4M1HNyivMthMGNr
bCxfM7QnShXgbI/xat96GqOKcDj950hae3qRQnFhoo1kBzLtNCvF7QYHgFJhVPo7
6jVv1jzeXZsgzq2/C+pM3rhZrXeHlNs2GBIQfeLe0mIHCP+xq1kDAJbboxdLtrWD
cmlgplpPL+18DNmPQ4MCtCJSOHHOeRppmXz5QdZO2sltHJeHVhyd3rFnCb3UjDPw
wLNvmSUEziGGrzCVI40Xt5jb2nCExRU2IAYiOVJy63BPtghSPAJ9BxCy+EEy/+NI
YY1G5dRXqT5LgApwaxjcBFrHKS1UlZRDtLNeHdKEaKMuO71rMffkA3fgDyruno6K
rKU2iF8b5LbSsw2Ra5Gj+PcuDY3kYV+ghacolNdGKhzsasGF0r9OpdPOwi2/TB8c
DEZKzFTb2iaE1YqxYA9rDBJB7g6vmno6vEkH6lWAAeJiEIo0m+IjsPpA2hz+WYeX
q3gvq+f0pVe5VsUA5WTI
=91IQ
-----END PGP SIGNATURE-----